src/main/java/org/elasticsearch/indices/recovery/RecoveryState.java

```diff
@@ -114,7 +115,7 @@ public static Type fromId(byte id) throws ElasticsearchIllegalArgumentException
         }
     }
 
-    private volatile Stage stage = Stage.INIT;
+    private volatile Stage stage;
 
```

Comment:
I'd feel better if we could remove the volatile and make the getter synced such that we never read a state while it's been updated?